### PR TITLE
Update doc for FinalParameters #7588

### LINKED ISSUE #7588: Update doc for FinalParameters
https://checkstyle.org/config_misc.html#FinalParameters

Add example for each existing config to show what code is ok and what code cause violation.
Good example is https://checkstyle.org/config_naming.html#AbstractClassName

please play with module/Check by means of https://checkstyle.org/cmdline.html to understand how it works.

------------

PLEASE READ https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-update-examples

### PR DESCRIPTION
**Pull request by Taniya003**
Fixes #7588

\> type config.xml
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="FinalParameters">
        </module>
    </module>
</module>

```
\> type Point.java
```
 public class Point {
    public Point() { } // ok
    public Point(final int m) { } // ok
    public Point(final int m,int n) { } // violation, n should be final
    public void methodOne(final int x) { } // ok
    public void methodTwo(int x) { } // violation, x should be final
    public static void main(String[] args) { } // violation, args should be final
  }
```
```
C:\Users\DELL\Desktop\tests>java -jar checkstyle-9.2.1-all.jar -c config.xml Point.java
Starting audit...
[ERROR] C:\Users\DELL\Desktop\tests\Point.java:11:30: Parameter n should be final. [FinalParameters]
[ERROR] C:\Users\DELL\Desktop\tests\Point.java:19:27: Parameter x should be final. [FinalParameters]
[ERROR] C:\Users\DELL\Desktop\tests\Point.java:23:29: Parameter args should be final. [FinalParameters]
Audit done.
Checkstyle ends with 3 errors.
```
\> type config.xml
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="FinalParameters">
        <property name="tokens" value="CTOR_DEF"/>
        </module>
    </module>
</module>
```
\> type Point.java
```
 public class Point {
    public Point() { } // ok
    public Point(final int m) { } // ok
    public Point(final int m,int n) { } // violation, n should be final
    public void methodOne(final int x) { } // ok
    public void methodTwo(int x) { } // ok
    public static void main(String[] args) { } // ok
 }
```
```
C:\Users\DELL\Desktop\tests>java -jar checkstyle-9.2.1-all.jar -c config.xml Point.java
Starting audit...
[ERROR] C:\Users\DELL\Desktop\tests\Point.java:11:30: Parameter n should be final. [FinalParameters]
Audit done.
Checkstyle ends with 1 errors.
```
\> type config.xml
```
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="FinalParameters">
        <property name="ignorePrimitiveTypes" value="true"/>
        </module>
    </module>
</module>
```
\> type Point.java
```
 public class Point {
    public Point() { } // ok
    public Point(final int m) { } // ok
    public Point(final int m,int n) { } // ok
    public void methodOne(final int x) { } // ok
    public void methodTwo(int x) { } // ok
    public static void main(String[] args) { } // violation, args should be final
  }
```
```
C:\Users\DELL\Desktop\tests>java -jar checkstyle-9.2.1-all.jar -c config.xml Point.java
Starting audit...
[ERROR] C:\Users\DELL\Desktop\tests\Point.java:23:29: Parameter args should be final. [FinalParameters]
Audit done.
Checkstyle ends with 1 errors.
```

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/FinalParametersCheck.java [hunk 1]
```diff
@@ -68,13 +68,71 @@
...
+ *       private int m,n;
+ *       private String s;
+ *       public Point(){ //ok
+ *       s = "ABC";
+ *       }
+ *       public Point(final int m) { // ok
+ *           this();
+ *           this.m = m;
+ *       }
+ *       public Point(final int m,int n){ //Violation - n should be final
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/FinalParametersCheck.java:84]: make space after `//` and lovercacse first word. Please apply this for all lines.

`// violation - n should be final`

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/FinalParametersCheck.java [hunk 2]
```diff
@@ -68,13 +68,71 @@
  * &lt;module name=&quot;FinalParameters&quot;/&gt;
  * </pre>
  * <p>
+ * Example :
+ * </p>
+ * <pre>
+ * public class Point {
+ *       private int m,n;
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/FinalParametersCheck.java:75]: please make indentation base +2 to keep code shorter at width.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/FinalParametersCheck.java [hunk 3]
```diff
@@ -68,13 +68,71 @@
...
+ *       private String s;
+ *       public Point(){ //ok
+ *       s = "ABC";
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/FinalParametersCheck.java:78]: +2 in indentation.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/FinalParametersCheck.java [hunk 4]
```diff
@@ -68,13 +68,71 @@
...
+ *       }
+ *       public Point(final int m) { // ok
+ *           this();
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/FinalParametersCheck.java:81]: This Check do not care about body content. so please make all metods single line
`public Point(final int m) { }`

**Review by romani — CHANGES_REQUESTED** [review]
review CIs and fix violations from them
make sure that `mvn verify` pass on your local in terminal before push.

items:

**Comment by Taniya003** [comment]
**@romani** I got all your points and followed the same. Thanks alot ;)

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/FinalParametersCheck.java [hunk 5]
```diff
@@ -68,6 +68,21 @@
  * &lt;module name=&quot;FinalParameters&quot;/&gt;
  * </pre>
  * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * public class Point {
+ *   private int m,n;
+ *   private String s;
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/FinalParametersCheck.java:76]: Remove fields
From all examples

**Review by romani — CHANGES_REQUESTED** [review]

**Comment by romani** [comment]
GitHub, rebase

**Comment by romani** [comment]
GitHub, generate web site

**Comment by romani** [comment]
@Taniya003, please rebase to most latest our master code. 
Your  update is awesome, let's do rebase, pass CI and merge.

**Comment by romani** [comment]
GitHub, generate web site

**Review by romani — APPROVED** [review]
ok to merge

**Review by strkkk — APPROVED** [review]

**Pull request merged**